resources/packs/googleworkspace/connected_apps.go

```diff
@@ -0,0 +1,113 @@
+package googleworkspace
+
+import (
+	"go.mondoo.com/cnquery/resources/packs/core"
+	"go.mondoo.com/cnquery/stringx"
+)
+
+type connectedApp struct {
+	clientID string
+	scopes   []string
+	name     string
+	appType  string
+	users    []*mqlGoogleworkspaceUser
+	tokens   []*mqlGoogleworkspaceToken
+}
+
+func (g *mqlGoogleworkspace) GetConnectedApps() ([]interface{}, error) {
+	// get all users
+	users, err := g.Users()
+	if err != nil {
+		return nil, err
+	}
+
+	connectedApps := map[string]*connectedApp{}
+	for _, user := range users {
+		usr := user.(*mqlGoogleworkspaceUser)
+		// get all token from user
+		tokens, err := usr.GetTokens()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, token := range tokens {
+			tk := token.(*mqlGoogleworkspaceToken)
+
+			clientID, err := tk.ClientId()
+			if err != nil {
+				return nil, err
+			}
+
+			cApp, ok := connectedApps[clientID]
+			if !ok {
+				cApp = &connectedApp{
+					clientID: clientID,
+					users:    []*mqlGoogleworkspaceUser{},
+					tokens:   []*mqlGoogleworkspaceToken{},
+				}
+			}
+
+			// assign name
+			displayText, err := tk.DisplayText()
+			if err != nil {
+				return nil, err
+			}
+			cApp.name = displayText
+
+			// merge scopes
+			scopes, err := tk.Scopes()
+			if err != nil {
+				return nil, err
+			}
+			stringScopes := []string{}
+			for _, scope := range scopes {
+				stringScopes = append(stringScopes, scope.(string))
+			}
+			cApp.scopes = stringx.DedupStringArray(append(cApp.scopes, stringScopes...))
+
+			cApp.tokens = append(cApp.tokens, tk)
+			cApp.users = append(cApp.users, usr)
+
+			connectedApps[clientID] = cApp
+		}
+	}
+
+	// group token by client id
+	runtime := g.MotorRuntime
+	res := []interface{}{}
```

Comment:
nit: could write that as `make([]interface{}, 0, len(connectedApps)`. That would initialize the underlying array with the correct size, instead of (potentially) moving and extending the current one after each iteration. Same for the slices below

Reply:
great idea, will adjust the PR